src/main/scala/io/iohk/ethereum/domain/Address.scala

```diff
@@ -0,0 +1,28 @@
+package io.iohk.ethereum.domain
+
+import akka.util.ByteString
+import io.iohk.ethereum.vm.DataWord
+
+case object Address {
+
+  val Length = 20
+
+  def apply(dw: DataWord): Address = Address(dw.bytes.takeRight(Length))
+
+  def apply(arr: Array[Byte]): Address = Address(ByteString(arr))
+
+  val empty: Address = Address(ByteString())
+
+}
+
+case class Address(bytes: ByteString) {
```

Comment:
Please note that a mutable `ArrayBuffer` is also a subclass of `IndexedSeq`. 
I'd prefer to be sure that we use ropes instead of plain byte arrays in our domain models. This makes sense especially in case of `Address`es. Thanks to `ByteString` implementation we avoid copying of memory eg. in line 10 in `Address.scala`.